### PR TITLE
feat: implement min and max for pyarrow

### PR DESCRIPTION
**Pull request by brentomagic**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
hi @brentomagic - looks like you added an empty file?

**Comment by brentomagic** [comment]
my bad, let me correct that.

**Comment by MarcoGorelli** [comment]
hey @brentomagic - looks like you're missing several imports, you should see if you can run this locally

**Comment by MarcoGorelli** [comment]
thanks @brentomagic ! the test runs, but now we get
```
E   AttributeError: 'ArrowNamespace' object has no attribute 'min'
```

Could you add that please? Take a look at `PandasLikeNamespace`, see how `min` is implemented there, and try to add it

File: tests/expr/reductions_test.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import narwhals.stable.v1 as nw
+from tests.utils import compare_dicts
+
+
+def test_expr_min_max(constructor_with_pyarrow: Any) -> None:
+    data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
+    df = nw.from_native(constructor_with_pyarrow(data), eager_only=True)
+    result_min = nw.to_native(df.select(nw.min("a", "b", "z")))
+    result_max = nw.to_native(df.select(nw.max("a", "b", "z")))
```
Comment (MarcoGorelli) [inline tests/expr/reductions_test.py:13]:
```suggestion
    result_min = nw.to_native(df.select(nw.col("a", "b", "z").min()))
    result_max = nw.to_native(df.select(nw.col("a", "b", "z").max()))
```

**Comment by MarcoGorelli** [comment]
looking good!!!

looks like you're just missing the implementation for `max`?
```
FAILED tests/expr/reductions_test.py::test_expr_min_max[table] - AttributeError: 'ArrowExpr' object has no attribute 'max'
```

File: narwhals/_arrow/expr.py [hunk 2]
```diff
@@ -135,3 +145,22 @@ def to_string(self, format: str) -> ArrowExpr:  # noqa: A002
...
+            "str",
+            "to_uppercase",
+        )
+
+    def to_lowercase(self) -> ArrowExpr:
+        return reuse_series_namespace_implementation(
+            self._expr,
+            "str",
+            "to_lowercase",
+        )
```
Comment (MarcoGorelli) [inline narwhals/_arrow/expr.py:166]: this looks irrelevant to your PR, I think it might have crept through in a merge conflict? could you clean up your branch so it only contains the changes you intended to make?

File: tests/expr/reductions_test.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import narwhals.stable.v1 as nw
+from tests.utils import compare_dicts
+
+
+def test_expr_min_max(constructor_with_pyarrow: Any) -> None:
+    data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
+    df = nw.from_native(constructor_with_pyarrow(data), eager_only=True)
+    result_min = nw.to_native(df.select(nw.col("a", "b", "z").min()))
+    result_max = nw.to_native(df.select(nw.col("a", "b", "z").max()))
```
Comment (MarcoGorelli) [inline tests/expr/reductions_test.py:13]: you can remove the `nw.to_native` part, then the test should pass

**Review by MarcoGorelli — COMMENTED** [review]
thanks @brentomagic for sticking to this, we're getting close!

**Review by MarcoGorelli — APPROVED** [review]
well done @brentomagic ! seriously good effort here!
